src/libmongoc/tests/unified/result.c

```diff
@@ -565,6 +567,43 @@ result_check (result_t *result,
             goto done;
          }
       }
+
+      if (error_response) {
+         if (!result->reply) {
+            test_set_error (
+               error, "%s", "expected error with a reply, but reply unset");
+            goto done;
+         }
+
+         /* Write operations have the raw response wrapped in a serverRespons
+          * document. Read operations return the raw response as the reply. */
+         bson_iter_t iter;
+         bson_t doc_to_match;
+         bson_init (&doc_to_match);
+
+         if (bson_iter_init_find (&iter, result->reply, "errorReplies")) {
+            bson_lookup_doc (result->reply, "errorReplies.0", &doc_to_match);
```

Comment:
Both `bson_lookup_doc` and `bson_look_up_doc_null_ok` will abort the test if the key is absent because they both use the helper function `bson_lookup` that will abort the test. However, I can still use `bson_lookup_doc_null_ok` to conditionally check if `errorReplies.0` was empty or not.

Reply:
Continuing this conversation in the current diff: https://github.com/mongodb/mongo-c-driver/pull/1144/files#r1022215846